### PR TITLE
User-Onboarding: Join community step

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6648

## Description of Changes
Added the 3rd step -> Join community step in the welcome onboard modal, which would suggest communities that match the auth user's chain network and allow to join those form the suggested list.

## "How We Fixed It"
N/A

## Test Plan
- Create a new account (this can be any account type)
- Login
- Apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts b/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
index 34e0f716ec..bf99934991 100644
--- a/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
+++ b/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
@@ -9,7 +9,7 @@ interface WelcomeOnboardModalProps {
 
 export const welcomeOnboardModal = createStore<WelcomeOnboardModalProps>()(
   devtools((set) => ({
-    isWelcomeOnboardModalOpen: false,
+    isWelcomeOnboardModalOpen: true,
     setIsWelcomeOnboardModalOpen: (isOpen) => {
       set((state) => {
         return {
diff --git a/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx b/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
index c3601ea350..cfb8dd6d7f 100644
--- a/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
+++ b/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
@@ -10,7 +10,7 @@ import { WelcomeOnboardModalProps, WelcomeOnboardModalSteps } from './types';
 
 const WelcomeOnboardModal = ({ isOpen, onClose }: WelcomeOnboardModalProps) => {
   const [activeStep, setActiveStep] = useState<WelcomeOnboardModalSteps>(
-    WelcomeOnboardModalSteps.PersonalInformation,
+    WelcomeOnboardModalSteps.JoinCommunity,
   );
```
- Verify you see the 3rd step of welcome modal (aka Join Communities)
- Verify that once you join a community, its `Join community` button changes to `Joined` and is not clickable anymore
- Verify that you can join all communities (for now we only suggest 4 communities)
- Verify that you can skip this step by clicking on `Let's go` or closing the modal.

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        isJoined:
+          suggestion.community.id === community.id ? true : suggestion.isJoined,
+      })),
+    ]);
+  };
+
+  return (
+    <section className="JoinCommunityStep">
+      <CWText type="h4" fontWeight="semiBold">
+        Based on your interests with think you&apos;ll like...
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx:61]: is this typo? with => we

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    setSuggestedCommunities(
+      chains.map((chain) => ({
+        community: chain,
+        isJoined: Permissions.isCommunityMember(chain.id),
+      })),
+    );
+  }, []);
+
+  const handleCommunityJoin = async (community: ChainInfo) => {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx:42]: can you use `useJoinCommunity` hook?

**Review by masvelio — CHANGES_REQUESTED** [review]
1. I am getting this form error as soon as I sign in and the modal appears. We should show the error only when the form is dirty && input is `!== ""`

image

2. On some pages the modal is appearing on refresh and on some pages it is not. Is it expected? see video below

3. When I generate the name, the `next` button is not enabled automatically. I need to type additional character to enable it. See video below

https://github.com/hicommonwealth/commonwealth/assets/14819225/2765b4f6-bf34-4f74-815c-c7ff61ee7843


4. Please always remember about mobile 

https://github.com/hicommonwealth/commonwealth/assets/14819225/618c9b78-05a0-4b59-aaf1-65a5a39cab44

5. You cant prolly see it on the video but when I click X or outside the modal, it disappears and appears quickly again, which effects in weird blink. Is it because of the applied patch?

https://github.com/hicommonwealth/commonwealth/assets/14819225/49140928-db08-4e14-b887-48b711c7eb2d


6. When I sign up from non-scoped page eg dashboard, I am automatically joined to the community:
- using metamask/evm -> ethereum
- using keplr => osmosis
Is this intentional? 

image

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx:61]: ah nice catch, I just copied it from figma and didn't recheck. Updated now.

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx:42]: actually we are using that, the `linkSpecificAddressToSpecificCommunity()` comes from `useJoinCommunity` hook

**Comment by mzparacha** [comment]
1. Wasn't able to repro consistently, but definitely flaky. Added the suggested fix, thanks for spotting.
2. I am not able to repro this. Tried this on `/communities` like in video, but it showed up for me. Logic wise the modal should show up on any page which has the `SubLayout` component and i believe nearly all components have it.
3. Fixed, thanks.
4. Fixed, thanks again for the spot.
5. If you applied the diff from OP and your account name is `Anonymous` then yes it's expected. It won't happen in prod since users cant jump to step 3 unless they finish step 1 and change their name to something other than `Anonymous`
6. Yes, this is the default behavior in prod. 

TLDR: can you recheck 2 if it consistently reproducible?


cc: @masvelio

**Review by masvelio — APPROVED** [review]

**Review by burtonator — APPROVED** [review]
